### DISCUSSION TITLE
Is there any config file like `nodemon` or `typescript`?

### DISCUSSION THREAD

**Discussion by nirvikpurkait** [prisma/prisma#23679] [Q&A] [answered]
### Question

I have a project with prisma implemented. For organizational purpose (or any other purpose), if we move the initialized folder to a different location (from its root location), for each script we have to explicitly provide the `schema.prisma` file location.

 Now if I have multiple scripts for different purpose I have to provide the path to each script. Such as :-

```ts
    // package.json

    "prisma-build": "npx prisma migrate deploy --schema=./src/database/prisma/schema.prisma",
    "prisma-dev": "npx prisma migrate dev --schema=./src/database/prisma/schema.prisma",
    "prisma-reset": "npx prisma migrate reset --schema=./src/database/prisma/schema.prisma",
    "prisma-studio": "npx prisma studio --schema=./src/database/prisma/schema.prisma",
    "prisma-push": "npx prisma db push --schema=./src/database/prisma/schema.prisma",
```

That looks ugly and repetitive.

Do we have a config file, like we have `nodemon.json` for **nodemon** or `tsconfig.json` for **typescript**, where we can put some arguments defining the path of the `schema.prisma` file, and each time we execute a command the source will be fetched from the config file.

**Comment by ludralph** [comment, marked as answer]
Hi @nirvikpurkait 👋 

Thank you for raising this question.

[Prisma CLI](https://www.prisma.io/docs/orm/reference/prisma-cli-reference#schema) provides a way to specify configuration settings using the `prisma` property in your package.json file. Within this property, you can define a `schema` key to specify the path to your `schema.prisma` file. This way, you don't have to provide the `--schema` flag for each command execution. The path defines the file the Prisma CLI should use when you run any of the CLI commands. Both absolute and relative paths are supported.

Here's how you can set up your package.json file:
```json
// package.json

{
  "prisma": {
    "schema": "./src/database/prisma/schema.prisma"
  },
  "scripts": {
    "prisma-build": "npx prisma migrate deploy",
    "prisma-dev": "npx prisma migrate dev",
    "prisma-reset": "npx prisma migrate reset",
    "prisma-studio": "npx prisma studio",
    "prisma-push": "npx prisma db push"
  }
}
```

Now, when you run any Prisma command, it will automatically use the schema file specified in the `prisma` property without the need to explicitly provide the path in each script.


If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nirvikpurkait** [reply]
Thanks a lot. I had this question in my mind for a long time.

**Reply by nirvikpurkait** [reply]
Hey @ludralph , Thanks for the previous answer.

 I was wandering is there any similar way of loading the `DATABASE_URL` form different `.env` files like -
 `.env.testing` - for testing locally
`.env.development` - for development
`.env.production` - for local production build, rather than the default `.env` file for all purpose.

N.B - I am using `Next.Js`, that is the reason this thought came into my mind as the `Next.Js` supports different env file for different purpose out of the box.

**Reply by ludralph** [reply]
Hi @nirvikpurkait 👋 

Prisma does not support this out of the box yet. We have documentation on how you can manually achieve that: https://www.prisma.io/docs/guides/development-environment/environment-variables/using-multiple-env-files
